### PR TITLE
[3.4] clientv3/balancer: fixed flaky TestRoundRobinBalancedResolvableFailoverFromServerFail

### PR DESCRIPTION
**Pull request by lavacat**
- ignore "transport is closing" error during connections warmup after stopping one peer.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by lavacat** [comment]
I couldn't reproduce failure locally, but looking at the logs and considering that this error was skipped in original test, I think this is the fix.

**Comment by serathius** [comment]
Could you confirm how this was fixed in main branch?

**Comment by lavacat** [comment]
@serathius N/A for main and 3.5 because balancer and picker packages were removed [here](https://github.com/etcd-io/etcd/pull/12671/)

File: clientv3/balancer/balancer_test.go [hunk 1]
```diff
@@ -301,15 +302,18 @@ func TestRoundRobinBalancedResolvableFailoverFromRequestFail(t *testing.T) {
...
 	// otherwise we can't reliably count switches between 'picked' peers in the test assert phase
 	for len(available) < serverCount {
 		picked, err = reqFunc(context.Background())
-		if err != nil {
+		if err != nil && ignoreErr != "" && strings.Contains(err.Error(), ignoreErr) {
+			// skip ignored errors
+			continue
+		} else if err != nil {
 			return available, picked, err
 		}
```
Comment (ahrtr) [inline clientv3/balancer/balancer_test.go:318]:
```suggestion
		if err != nil {
		    if  ignoreErr != "" && strings.Contains(err.Error(), ignoreErr) {
			    // skip ignored errors
			    continue
		    }
		    return  available, picked, err
		} 
```

**Comment by ahrtr** [comment]
Please fix the fmt error. 

```
gofmt -w the_file_you_changed.
```

**Review by ahrtr — APPROVED** [review]
LGTM

**Pull request merged**